From af460bd8b098ba7617f2a5b76e4c36916c397700 Mon Sep 17 00:00:00 2001
From: Zheng Yang <zhengyang@rock-chips.com>
Date: Tue, 4 Aug 2015 18:07:54 +0800
Subject: [PATCH] hdmi:rk3288/rk3368: Reset tmdsclk after configure frame
 composer regiseter.

It is required to perform a reset tmdsclk action on one of the frame composer
registers changed. Or transport video and audio sample may mistake.

Signed-off-by: Zheng Yang <zhengyang@rock-chips.com>
---
 .../video/rockchip/hdmi/rockchip-hdmiv2/rockchip_hdmiv2_hw.c   | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/hdmi/rockchip-hdmiv2/rockchip_hdmiv2_hw.c b/drivers/video/rockchip/hdmi/rockchip-hdmiv2/rockchip_hdmiv2_hw.c
index 46023e399233..c1120f78c8cf 100755
--- a/drivers/video/rockchip/hdmi/rockchip-hdmiv2/rockchip_hdmiv2_hw.c
+++ b/drivers/video/rockchip/hdmi/rockchip-hdmiv2/rockchip_hdmiv2_hw.c
@@ -685,7 +685,8 @@ static int rockchip_hdmiv2_video_framecomposer(struct hdmi *hdmi_drv,
 	 */
 	hdmi_writel(hdmi_dev, FC_EXCTRLSPAC,
 		    (hdmi_dev->tmdsclk/1000) * 50 / (256 * 512));
-
+	hdmi_msk_reg(hdmi_dev, MC_SWRSTZREQ,
+		     m_TMDS_SWRST, v_TMDS_SWRST(0));
 #if 0
 	/*Set PreambleFilter*/
 	for (i = 0; i < 3; i++) {
-- 
2.35.3

